components/automate-ui/src/app/components/projects-dropdown/projects-dropdown.component.html

```diff
@@ -14,15 +14,34 @@
   </button>
   <chef-click-outside (clickOutside)="closeDropdown()">
     <chef-dropdown [attr.visible]="active">
-      <chef-checkbox
-        *ngFor="let project of projectsArray"
-        [checked]="project.checked"
-        [attr.title]="project.name"
-        (change)="projectChecked($event.detail, project)"
-        (keydown.enter)="closeDropdown()"
-        (keydown.esc)="closeDropdown()"
-        (keydown.arrowup)="moveFocus($event)"
-        (keydown.arrowdown)="moveFocus($event)">{{ project['name'] }}</chef-checkbox>
+
+      <div id="filter-container">
+        <input 
+          chefInput
+          type="text"
+          [(ngModel)]="filterValue"
+          placeholder="Filter projects..." 
+          (keyup)="handleFilterKeyUp()"
+          />
+      </div>
+
+      <div id="project-container">
+        <chef-checkbox
+          *ngFor="let project of filteredProjects"
+          [checked]="project.checked"
+          [attr.title]="project.name"
+          (change)="projectChecked($event.detail, project)"
+          (keydown.enter)="closeDropdown()"
+          (keydown.esc)="closeDropdown()"
+          (keydown.arrowup)="moveFocus($event)"
+          (keydown.arrowdown)="moveFocus($event)">{{ project['name'] }}</chef-checkbox>
```

Comment:
Suggest use properties instead of magic strings when possible--I think that should work here:
```suggestion
          (keydown.arrowdown)="moveFocus($event)">{{ project.name }}</chef-checkbox>
```